### ISSUE TITLE
adding medical specialities

### ISSUE THREAD

**Issue by matkoniecz** [Helium314/SCEE#408]
Not exactly sure what kind of changes you want but picking up https://github.com/matkoniecz/Zazolc/commit/ca3ecde84ef41792f16993818cbcc26eac0213cc (adding medical specialities) may be useful if you or other users would report missing specialities in iD presets and/or contribute translations to iD presets in their languages, making this quest more ready for wider release and testing is it even a really good idea.

Note that you can add only one at time, so multiple ones would require making a note.

So far as result of testing this code I opened https://github.com/openstreetmap/id-tagging-schema/issues/778 https://github.com/openstreetmap/id-tagging-schema/issues/779 https://www.openstreetmap.org/changeset/132128903 https://www.openstreetmap.org/note/3546322 (pain management - it seems we have no tag for that) when I tried to use it

(StreetComplete issue is at https://github.com/streetcomplete/StreetComplete/issues/1020 )

**Comment by Helium314** [comment]
This would definitely be useful, though I think I'll adapt the quest to fit into the current healthcare speciality quest (and allow switching to the current UI for entering multiple values).

**Comment by matkoniecz** [comment]
ooo, you have code for that? Is it problematic in some way or unfinished (if it is working well then feel free to make SC PR!)?

**Comment by matkoniecz** [comment]
> First thing noticed missing: dermatovenereology

If you are sure that it should be added and it has OSM tag then reporting it at https://github.com/openstreetmap/id-tagging-schema/issues would be the best idea (or opening PR there)

**Comment by Helium314** [comment]
Another thing is that it might be good to allow searching in language of the current country, so even if you're somewhere else you'd just need to enter what you see on the sign.
But it looks like the feature dictionary doesn't support that.

**Comment by matkoniecz** [comment]
>  But it looks like the feature dictionary doesn't support that (or I'm doing it wrong).

Note opening hours days of week display where SC got such feature (after field testing in Italy during STOTM)

**Comment by westnordost** [comment]
> First thing noticed missing: dermatovenereology

What's that, a doctor that specializes in .... German: "Sexuell über die Haut übertragbare Krankheiten"?
Maybe this field exists, but so do 1000s more. I recently added presets for common physician specializations in https://github.com/openstreetmap/id-tagging-schema/pull/718 but dermatovenereology doesn't sound common.

**Comment by Helium314** [comment]
> Maybe this field exists, but so do 1000s more

At least in Austria, "Facharzt für Haut- und Geschlechtskrankheiten" is rather common. Not sure, but maybe it's even some kind of default for dermatologist training.

**Comment by westnordost** [comment]
Oh well, if it is in the name... maybe consider creating a pr over there. I think i never read this speciality

Am 6. Februar 2023 21:17:06 MEZ schrieb Helium314 ***@***.***>:
>> Maybe this field exists, but so do 1000s more
>
>
>
>At least in Austria, "Facharzt für Haut- und Geschlechtskrankheiten" is rather common. Not sure, but maybe it's even some kind of default for dermatologist training.
>
>
>
>-- >
>Reply to this email directly or view it on GitHub:
>
>https://github.com/Helium314/SCEE/issues/408#issuecomment-1419692395
>
>You are receiving this because you commented.
>
>
>
>Message ID: ***@***.***>
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by Helium314** [comment]
Apparently this is connected also in Gemany: https://flexikon.doccheck.com/de/Dermatologie (Facharztweiterbildung), and [wikipedia](https://en.wikipedia.org/wiki/Venereology) claims that "In many areas of the world, the specialty is usually combined with dematology".

So I'm not sure now whether the distinction between `dermatology` and `dermatovenereology` really matters. Probably only in the some areas of the world.

**Comment by westnordost** [comment]
Also, I wonder if then it shouldn't be two specialities? `dermatology; venereology` ... anyway, I haven't had a look at your implementation, but that it is valid to have several specialities is something to consider.

Question: For your implementation, do you use the translatable field `healthcare:speciality` or the preset name? Or both? I think there are not aliases/terms for field values, which means that if you search for "Haut" in German you won't find `dermatology`. On the other hand, the list of translatable fields for `healthcare:speciality` in the iD presets is larger, because in my recent PR I only added the most common specialities as own presets.
In any case, if you use the translatable fields, how do you do it? IIRC https://github.com/westnordost/osmfeatures/ does not support (translatable) fields yet.

**Comment by Helium314** [comment]
> Also, I wonder if then it shouldn't be two specialities? 

Maybe it should actually be both, yes. I wouldn't consider this as a single one, if it didn't already exist in the `[wiki](https://wiki.openstreetmap.org/wiki/Key:healthcare:speciality)`.

Until last version, the quest asked for plain OSM tags in an EditText, allowing only to enter tags from a pre-defined list (with suggestions), and multiple values.
With https://github.com/Helium314/SCEE/commit/3204e883a07e0fbb3e26a3e2c365f683702fc8e4 it uses the UI as it is currently planned for https://github.com/streetcomplete/StreetComplete/issues/1020 (plus optional fallback to old UI for entering multiple values).
Basically it's the default `SearchFeaturesDialog`, with filtering for doctors. Here you can enter `skin` and it will find _Dermatologist_. Works similar in German.

**Comment by matkoniecz** [comment]
Though my interface works only when things are defined as features (which triggered my iD issue - it has bad effects for iD mappers anyway so can be fixed also there) and still does not support multiple values.

I wanted to check is there any unexpected blocker before starting fight with RecyclerView.
